core/src/main/scala/scalaz/NonEmptyList.scala

```diff
@@ -238,6 +238,11 @@ sealed abstract class NonEmptyListInstances extends NonEmptyListInstances0 {
         (BindRec[IList].tailrecM[A, B](a)(a => f(a).list): @unchecked) match {
           case ICons(h, t) => NonEmptyList.nel(h, t)
         }
+
+      override def traverseImpl[F[_], A, B](fa: NonEmptyList[A])(f: A => F[B])(implicit F: Applicative[F]): F[NonEmptyList[B]] = {
+        val fb = Traverse[IList].traverseImpl(fa.list)(f)(F)
+        F.map(fb)(fb => NonEmptyList.nel(fb.headOption.getOrElse(sys.error("Head can't be empty")), fb.tailOption.getOrElse(INil())))        
```

Comment:
Using `sys.error` here was what first came to mind but let me know if there's another means of handling or message that is preferred.

Reply:
I would write a dedicated implementation for `NonEmptyList` (i.e. not delegate to `IList`).

Also, let's do that in the implementation of `traverse1` and then it suffices to keep the default `traverseImpl`.